onnxruntime/core/graph/contrib_ops/shape_inference_functions.cc

```diff
@@ -97,9 +99,11 @@ void EmbedLayerNormalizationShapeInference(::ONNX_NAMESPACE::InferenceContext& c
   updateOutputShape(ctx, 0, output_shape);
 
   // mask_index shape is (batch_size)
-  ONNX_NAMESPACE::TensorShapeProto mask_index_shape;
-  *mask_index_shape.add_dim() = input_ids_dims[0];
-  updateOutputShape(ctx, 1, mask_index_shape);
+  if (ctx.getNumOutputs() > 1) {
+    ONNX_NAMESPACE::TensorShapeProto mask_index_shape;
+    *mask_index_shape.add_dim() = input_ids_dims[0];
+    updateOutputShape(ctx, 1, mask_index_shape);
+  }
 
   if (ctx.getNumOutputs() > 2) {
```

Comment:
May need update logic here if we want to support outputs output + embedding_sum (mask_index is optional)

Reply:
Updated this check to see if the mask input shape exists

Reply:
Could we use ctx.getOutputType(1) == nullptr to check whether the output exists?